engines/sherlock/objects.cpp

```diff
@@ -0,0 +1,1131 @@
+	_flags = s.readByte();
+	_aOpen.load(s);
+	_aType = (AType)s.readByte();
+	_lookFrames = s.readByte();
+	_seqCounter = s.readByte();
+	_lookPosition.x = s.readUint16LE();
+	_lookPosition.y = s.readByte();
+	_lookFacing = s.readByte();
+	_lookcAnim = s.readByte();
+	_aClose.load(s);
+	_seqStack = s.readByte();
+	_seqTo = s.readByte();
+	_descOffset = s.readUint16LE();
+	_seqCounter2 = s.readByte();
+	_seqSize = s.readUint16LE();
+	s.skip(1);
+	_aMove.load(s);
+	s.skip(8);
+
+	for (int idx = 0; idx < 4; ++idx)
+		_use[idx].load(s);
+}
+
+/**
+ * Toggle the type of an object between hidden and active
+ */
+void Object::toggleHidden() {
+	if (_type != HIDDEN && _type != HIDE_SHAPE && _type != INVALID) {
+		if (_seqTo != 0)
+			_sequences[_frameNumber] = _seqTo + SEQ_TO_CODE + 128;
+		_seqTo = 0;
+
+		if (_images == nullptr || _images->size() == 0)
+			// No shape to erase, so flag as hidden
+			_type = HIDDEN;
+		else
+			// Otherwise, flag it to be hidden after it gets erased
+			_type = HIDE_SHAPE;
+	} else if (_type != INVALID) {
+		if (_seqTo != 0)
+			_sequences[_frameNumber] = _seqTo + SEQ_TO_CODE + 128;
+		_seqTo = 0;
+
+		_seqCounter = _seqCounter2 = 0;
+		_seqStack = 0;
+		_frameNumber = -1;
+
+		if (_images == nullptr || _images->size() == 0) {
+			_type = NO_SHAPE;
+		} else {
+			_type = ACTIVE_BG_SHAPE;
+			int idx = _sequences[0];
+			if (idx >= _maxFrames)
+				// Turn on: set up first frame
+				idx = 0;
+
+			_imageFrame = &(*_images)[idx];
+		}
+	}
+}
+
+/**
+ * Check the state of the object
+ */
+void Object::checkObject() {
+	Scene &scene = *_vm->_scene;
+	Sound &sound = *_vm->_sound;
+	int checkFrame = _allow ? MAX_FRAME : 32000;
+	bool codeFound;
+
+	if (_seqTo) {
+		byte *ptr = &_sequences[_frameNumber];
+		if (*ptr == _seqTo) {
+			// The sequence is completed
+			*ptr = _seqTo + SEQ_TO_CODE + 128;	// Reset to normal
+			_seqTo = 0;
+		} else {
+			// Continue doing sequence
+			if (*ptr > _seqTo)
+				*ptr -= 1;
+			else
+				*ptr += 1;
+
+			return;
+		}
+	}
+
+	++_frameNumber;
+
+	do {
+		// Check for end of sequence
+		codeFound = checkEndOfSequence();
+
+		if (_sequences[_frameNumber] >= 128 && _frameNumber < checkFrame) {
+			codeFound = true;
+			int v = _sequences[_frameNumber];
+
+			if (v >= 228) {
+				// Goto code found
+				v -= 228;
+				_seqCounter2 = _seqCounter;
+				_seqStack = _frameNumber + 1;
+				setObjSequence(v, false);
+			} else if (v >= SOUND_CODE && (v <= (SOUND_CODE + 29))) {
+				codeFound = true;
+				++_frameNumber;
+				v -= SOUND_CODE;
+
+				if (sound._soundOn && !_countCAnimFrames) {
+					if (!scene._sounds[v - 1]._name.empty() && sound._digitized)
+						sound.playLoadedSound(v - 1, WAIT_RETURN_IMMEDIATELY);
+				}
+			} else if (v >= FLIP_CODE && v <= (FLIP_CODE + 2)) {
+				// Flip code
+				codeFound = true;
+				++_frameNumber;
+				v -= FLIP_CODE;
+
+				// Alter the flipped status
+				switch (v) {
+				case 0:
+					// Clear the flag
+					_flags &= ~2;
```

Comment:
Again, it is probably a good idea to introduce constant names here for all the possible flags.